pkg/armhelpers/azureclient.go

```diff
@@ -462,3 +463,45 @@ func (az *AzureClient) addAcceptLanguages() autorest.PrepareDecorator {
 		})
 	}
 }
+
+func (az *AzureClient) addAuxiliaryTokens() autorest.PrepareDecorator {
+	return func(p autorest.Preparer) autorest.Preparer {
+		return autorest.PreparerFunc(func(r *http.Request) (*http.Request, error) {
+			r, err := p.Prepare(r)
+			if err != nil {
+				return r, err
+			}
+			if az.auxiliaryTokens != nil {
+				for _, token := range az.auxiliaryTokens {
+					if token == "" {
+						continue
+					}
+
+					r.Header.Set("x-ms-authorization-auxiliary", fmt.Sprintf("Bearer %s", token))
+				}
+			}
+			return r, nil
+		})
+	}
+}
+
+// AddAuxillaryTokens sets the list of aux tokens to accept on this request
+func (az *AzureClient) AddAuxillaryTokens(tokens []string) {
```

Comment:
Is this function going to be called in the RP? is it possible to add unit tests for it?